app/src/firebase/snaps/SnapService.ts

```diff
@@ -15,22 +15,26 @@ function docToSnap(data: any) {
     };
 }
 
-async function getSubmittedSnaps(user: firebase.User): Promise<Snap[]> {
-    try {
-        const userSnaps = await firebase
-            .firestore()
-            .collection("snaps")
-            .where("from", "==", user.uid.toString())
-            .orderBy("timestamp", "desc")
-            .get();
-        return userSnaps.docs.map((doc) => docToSnap(doc.data()));
-    } catch (error) {
-        throw new Error(error.message);
-    }
-}
-
-export async function getSubmittedSnapsForCurrentUser(): Promise<Snap[]> {
-    const currentUser = firebase.auth().currentUser;
-    const snaps = await getSubmittedSnaps(currentUser);
-    return snaps;
+// returns an unsubscribe function
```

Comment:
To take advantage of this you'll want to us the cleanup part of `useEffect`. (See [docs](https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1)).